src/ui/public/vis/vis.js

```diff
@@ -81,18 +81,23 @@ export default function VisFactory(Notifier, Private) {
     this.aggs = new AggConfigs(this, state.aggs);
   };
 
-  Vis.prototype.getState = function () {
+  Vis.prototype.getState = function (full) {
     return {
       title: this.title,
       type: this.type.name,
       params: this.params,
-      aggs: this.aggs.map(function (agg) {
-        return agg.toJSON();
-      }).filter(Boolean),
+      aggs: this.aggs
+        .filter(agg => full || agg.enabled)
+        .map(agg => agg.toJSON())
+        .filter(Boolean),
       listeners: this.listeners
     };
   };
 
+  Vis.prototype.getFullState = function () {
```

Comment:
Can we reverse this logic so that we're not introducing BC breaks to `getState`? So `getState` would continue to return everything whereas perhaps `getEnabledState` would return the state with only enabled aggregations?

We probably *can* change the behavior of `getState` without breaking anything, but bugs happen when we make assumptions.  Even if our assumptions are correct that this is OK for Kibana core, there are a bunch of plugins out there that add visualizations to Kibana - how confident can we be that this won't break them?